File: GetPlatform.ts
"use strict";
import * as os from "os";

export class OS{
	public static readonly platform:string=process.platform;
	public static readonly osName:string=os.type().toString();
	public static readonly version:string=os.release();
	public static readonly fullName:string=`${OS.osName} ${OS.version}`;
	public static readonly arch:string=process.env.PROCESSOR_ARCHITEW6432||process.arch;
	public static readonly fullNameArch:string=`${OS.fullName} (${OS.arch})`;
}

export class Runtime{
	public static readonly lang:string="TypeScript";
	public static readonly runtime:string="Node.js";
	public static readonly version:string=process.version;
	public static readonly arch:string=process.arch;
	public static readonly fullName:string=`${Runtime.lang}/${Runtime.runtime} ${Runtime.version}`;
	public static readonly fullNameArch:string=`${Runtime.fullName} (${Runtime.arch})`;
}

File: GetPlatform.cs
using System;
using System.Runtime.InteropServices;

namespace GetPlatform{
	class OS{
		public static readonly string platform=Environment.OSVersion.Platform.ToString();
		public static readonly string version=Environment.OSVersion.Version.ToString();
		public static readonly string fullName=Environment.OSVersion.VersionString;
		public static readonly string osName=OS.fullName.Substring(0,OS.fullName.IndexOf(OS.version)).Trim();
		public static readonly string arch=RuntimeInformation.OSArchitecture.ToString();
		public static readonly string fullNameArch=$"{OS.fullName} ({OS.arch})";
	}

	class Runtime{
		private static string rf=RuntimeInformation.FrameworkDescription;
		public const string lang="C#";
		public static readonly string runtime=rf.Substring(0,rf.LastIndexOf(" ")).Trim();
		public static readonly string version=rf.Substring(rf.LastIndexOf(" ")).Trim();
		public static readonly string fullName=$"{Runtime.lang}/{Runtime.rf}";
		public static readonly string arch=RuntimeInformation.ProcessArchitecture.ToString();
		public static readonly string fullNameArch=$"{Runtime.fullName} ({Runtime.arch})";
	}
}